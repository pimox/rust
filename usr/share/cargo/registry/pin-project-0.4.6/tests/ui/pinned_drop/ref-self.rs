use std::pin::Pin;

pub struct Foo {
    field: u8,
}

impl Foo {
    fn method_ref(ref self: Pin<&mut Self>) {} //~ ERROR expected identifier, found keyword `self`
    fn method_ref_mut(ref mut self: Pin<&mut Self>) {} //~ ERROR expected identifier, found keyword `self`
}

fn main() {}
