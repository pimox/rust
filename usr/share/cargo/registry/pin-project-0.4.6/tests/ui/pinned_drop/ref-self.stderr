error: expected identifier, found keyword `self`
 --> $DIR/ref-self.rs:8:23
  |
8 |     fn method_ref(ref self: Pin<&mut Self>) {} //~ ERROR expected identifier, found keyword `self`
  |                       ^^^^ expected identifier, found keyword

error: expected identifier, found keyword `self`
 --> $DIR/ref-self.rs:9:31
  |
9 |     fn method_ref_mut(ref mut self: Pin<&mut Self>) {} //~ ERROR expected identifier, found keyword `self`
  |                               ^^^^ expected identifier, found keyword
